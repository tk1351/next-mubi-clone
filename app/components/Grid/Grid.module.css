.grid {
    display: grid;
    grid-gap: 2rem;
    grid-template-columns: repeat(auto-fit, minmax(min(300px, 100%), 1fr));
}
