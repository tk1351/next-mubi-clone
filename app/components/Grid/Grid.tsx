import { type FC, type PropsWithChildren } from "react";
import styles from "./Grid.module.css";

export type Props = PropsWithChildren<{}>;

export const Grid: FC<Props> = ({ children }) => {
  return <div className={styles.grid}>{children}</div>;
};
