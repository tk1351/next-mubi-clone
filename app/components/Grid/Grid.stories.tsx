import type { Meta, StoryObj } from "@storybook/react";
import { Grid } from "./Grid";

const meta = {
  title: "Components/Grid",
  component: Grid,
} satisfies Meta<typeof Grid>;

export default meta;
type Story = StoryObj<typeof meta>;
export const Primary: Story = {
  args: {
    children: (
      <>
        {Array.from({ length: 10 }, (_, index) => (
          <div
            key={index}
            style={{
              height: "200px",
              width: "300px",
              border: "1px solid orange",
            }}
          ></div>
        ))}
      </>
    ),
  },
};
