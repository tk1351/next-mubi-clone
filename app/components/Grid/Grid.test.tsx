import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { Grid } from "./Grid";

describe(`Testing Grid Component`, () => {
  it("Snapshot", () => {
    const { container } = render(<Grid>{"Grid"}</Grid>);
    expect(container).toMatchSnapshot();
  });
});
