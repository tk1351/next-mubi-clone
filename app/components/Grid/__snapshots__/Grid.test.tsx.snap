// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing Grid Component > Snapshot 1`] = `
<div>
  <div
    class="_grid_296888"
  >
    Grid
  </div>
</div>
`;
